kpet/data.py

```diff
@@ -167,49 +200,59 @@ def __node_matches(self, target, and_op, node, qualifier):
                         Cannot be None if node is a None or a regex.
 
         Returns:
-            True if the node matched, False otherwise.
+            True if the node matched, False if not,
+            and None if the result could be any.
         """
         assert isinstance(target, Target)
         assert qualifier is None or qualifier in self.qualifiers
 
+        def sub_op(result_x, result_y):
+            """Combine two sub-results using the specified operation"""
+            if result_x is None:
+                return result_y
+            if result_y is None:
```

Comment:
I'm not completely sure what these results are, so I'm asking just in case: should these `None` be `ANY`? Is the result of the pattern matching operation a set like that?

Reply:
ANY should be None and I agree using ANY here might make it easier to read (and getting rid of a 'magic constant')

Reply:
Nope. These are not "target sets" here, these are `__node_matches()` return values which are defined as `True`/`False`/`None`.

Reply:
Thanks for checking, though!